src/Grace/DependencyInjection/IDisposalScope.cs

```diff
@@ -5,20 +5,42 @@ namespace Grace.DependencyInjection
     /// <summary>
     /// Represents a scope that holds disposable object
     /// </summary>
-    public interface IDisposalScope : IDisposable
+    public interface IDisposalScope
+#if NETSTANDARD2_1
+        : IDisposable, IAsyncDisposable
+#else
+        : IDisposable
+#endif
     {
+#if NETSTANDARD2_1
+        /// <summary>
+        /// Add an object for disposal tracking
+        /// </summary>
+        /// <param name="disposable">object to track for async disposal</param>
+        T AddAsyncDisposable<T>(T disposable)
+            where T : IAsyncDisposable;
 
+        /// <summary>
+        /// Add an object for disposal tracking
+        /// </summary>
+        /// <param name="disposable">object to track for async disposal</param>
+        /// <param name="cleanupDelegate">logic that will be run directly before the object is disposed</param>
+        T AddAsyncDisposable<T>(T disposable, Action<T> cleanupDelegate)
```

Comment:
This probably needs be `Func<T,Task>` as the cleanup action for an `IAsyncDisposable` is probably also async.

Reply:
I've avoided this change as this would require changes up to the IFluentExportStrategyConfiguration interface.
For now, I was thinking of skipping this to simplify current PR and add in the subsequent versions in case of requests from Grace's clients.

Reply:
For the moment we can leave as it does have a ripple effect all the way out to the configuration. I honestly don't know how much this is actually used so I don't know if it's worth plumbing `Func<T,Task>`